### ISSUE TITLE
Enable VS macros for build commands in properties 

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#4109]
As reported by @licinioamendes  in https://github.com/Microsoft/TypeScript/issues/2152#issuecomment-126757495:

> Would be nice allow [VS macros for build commands](https://msdn.microsoft.com/en-us/library/c02as0cs.aspx "MSDN Macros for Build Commands").

> The result would be something like `$(ProjectDir)outputDirectoryPath` on textbox.

Also note that this is working in VS2013 but not in VS2015.

**Comment by mhegazy** [comment]
Also note, this is causing the issue outlined in #4252.  courtesy of @NoelAbrahams 


* Set `"Redirect JavaScript output to directory"` to say `bin`.
* Set `"Generate declarations files"`.
* Set `"Compile on save"`.
* Add file `_references.ts` to the project, with content

 ```xml
 /// <reference path='../../foo/bar1/bin/bar1.d.ts' />
 /// <reference path='../../foo/bar2/bin/bar2.d.ts' />
 ``` 

* Build the project and check the output directory. The content of `_references.d.ts` should be

 ```xml
 /// <reference path="../../../foo/bar1/bin/bar1.d.ts' />
 /// <reference path="../../../foo/bar1/bin/bar2.d.ts' />
 ```

* Now modify one file in the project and trigger `compile-on-save` by saving the file. Now the content of the same file should have changed to 

 ```xml
 /// <reference path="d:/x/y/z/foo/bar1/bin/bar1.d.ts' />
 /// <reference path="d:/x/y/z/foo/bar1/bin/bar2.d.ts' />
 ```

This together with #4245 appears to be a regression introduced in the fix for #3939.

This is interfering with some systems we have in place.

**Comment by paulvanbrenk** [comment]
This will be in the next TS release for VS 2015
